### PR TITLE
Restructure shared http code

### PR DESCRIPTION
**Pull request by Tratcher**
I've restructured the shared HTTP/2 and HTTP/3 code as discussed. This moves both HTTP/2 and HTTP/3 code into a single shared folder to make keeping it in sync easier.

@jkotalik I've rolled in the changes from your PRs to make everything build on both sides.

Followup: rename the "http2" group to "http" and remove the "http3" group. The two groups contain the same people. Update Codeowners as needed.

See https://github.com/dotnet/aspnetcore/pull/18967 for the aspnetcore changes.

### CODE REVIEW

File: src/libraries/Common/src/System/Net/Http/aspnetcore/Http3/QPack/HeaderField.cs [hunk 1]
```diff
@@ -8,8 +8,8 @@ namespace System.Net.Http.QPack
     {
         public HeaderField(byte[] name, byte[] value)
         {
-            Name = new byte[name.Length];
```
Comment (jkotalik) [inline src/libraries/Common/src/System/Net/Http/aspnetcore/Http3/QPack/HeaderField.cs]: Revert this

**Review by jkotalik — APPROVED** [review]

File: src/libraries/Common/src/System/Net/Http/aspnetcore/Http3/QPack/HeaderField.cs (hunk 1 above)
Comment (jkotalik) [reply src/libraries/Common/src/System/Net/Http/aspnetcore/Http3/QPack/HeaderField.cs]: wait nvrm, keep it.

**Review by scalablecory — APPROVED** [review]
Thanks for this. Please let CI finish before merge.

**Comment by Tratcher** [comment]
Unrelated failures in NetworkInformation and Tasks. Merging.

**Pull request merged**